backend/bin/kafkaConsumer/common/userCourseProgress/checkAndSendThresholdEmail.ts

```diff
@@ -0,0 +1,63 @@
+require("dotenv-safe").config({
+  allowEmptyValues: process.env.NODE_ENV === "production",
+})
+import { User, Course } from "@prisma/client"
+import { CombinedUserCourseProgress } from "./userFunctions"
+import { KafkaContext } from "../kafkaContext"
+import { v4 as uuidv4 } from "uuid"
+
+interface Props {
+  user: User
+  course: Course
+  combinedUserCourseProgress: CombinedUserCourseProgress
+  context: KafkaContext
+}
+
+export const checkAndSendThresholdEmail = async ({
+  user,
+  course,
+  combinedUserCourseProgress,
+  context,
+}: Props) => {
+  const courseEmailThresholdTemplates = await context.prisma.emailTemplate.findMany(
+    {
+      where: {
+        triggered_automatically_by_course_id: course.id,
+        template_type: "threshold",
+      },
+    },
+  )
+
+  const templatesThatFulfillPoints = courseEmailThresholdTemplates
+    .sort((a, b) => (b.points_threshold ?? 0) - (a.points_threshold ?? 0))
+    .filter((et) => {
+      return (
+        combinedUserCourseProgress.total_n_points >=
+        (et.points_threshold ?? 99999)
```

Comment:
All threshold templates will trigger with this when total_n_points is over 99999 (which can happen for example accidentally). One way to fix this could be to use the global `Infinity` instead of 99999 but, for clarity's sake, I'd prefer here an explicit null check and an early return.

Here's one possible example:
```js
if (et.points_threshold === null || et.points_threshold === undefined) {
  return false;
}
return combinedUserCourseProgress.total_n_points >= et.points_threshold
```

Reply:
Fixed with early return.